src/components/accordion/accordion.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { Plus, Minus, ChevronDown } from 'lucide-react';
+import { cn } from '@/utilities/functions';
+
+const Accordion = ( props ) => {
+	const {
+		type = 'simple',
+		defaultValue,
+		disabled = false,
+		children,
+		className,
+	} = props;
+	const [ activeItem, setActiveItem ] = useState( defaultValue );
+
+	const handleToggle = ( value ) => {
+		setActiveItem( ( prev ) => ( prev === value ? null : value ) );
+	};
+
+	const typeClasses = type === 'boxed' ? 'space-y-3' : '';
+
+	return (
+		<div className={ cn( typeClasses, className ) }>
+			{ React.Children.map( children, ( child ) =>
+				React.cloneElement( child, {
+					isActive: child.props.value === activeItem,
+					onToggle: () => handleToggle( child.props.value ),
+					type,
+					disabled: disabled || child.props.disabled,
+				} )
+			) }
+		</div>
+	);
+};
+
+Accordion.displayName = 'Accordion';
+
+const AccordionItem = ( props ) => {
+	const {
+		isActive,
+		onToggle,
+		type,
+		value,
+		disabled = false,
+		children,
+		className,
+	} = props;
+
+	let typeClasses = 'border-0';
+	if ( type === 'separator' ) {
+		typeClasses = 'border-0 border-b border-solid border-border-subtle';
+	} else if ( type === 'boxed' ) {
+		typeClasses = 'border border-solid border-border-subtle rounded-md';
+	}
+
+	return (
+		<div className={ cn( typeClasses, className ) } value={ value }>
+			{ React.Children.map( children, ( child ) =>
+				React.cloneElement( child, { isActive, onToggle, disabled } )
+			) }
+		</div>
+	);
+};
+
+AccordionItem.displayName = 'Accordion.Item';
+
+const AccordionTrigger = ( props ) => {
+	const {
+		onToggle,
+		isActive,
+		iconType, // arrow, plus-minus
+		disabled = false,
+		tag = 'h3',
+		children,
+		className,
+	} = props;
+
+	const renderIcon = () => {
+		if ( iconType === 'arrow' ) {
+			return (
+				<ChevronDown
+					className={ cn(
+						'text-icon-secondary transition-transform duration-500 ease-in-out',
+						isActive ? 'rotate-180' : 'rotate-0'
+					) }
+				/>
+			);
+		}
+		if ( iconType === 'plus-minus' ) {
+			return isActive ? (
+				<Minus className="text-icon-secondary" />
+			) : (
+				<Plus className="text-icon-secondary" />
+			);
+		}
+		return null;
+	};
+
+	const Tag = tag;
+	return (
+		<Tag className="flex m-0">
+			<button
```

Comment:
We can add ARIA roles from now

Reply:
From w3school resources: `A <button> has the role="button" built in, so to write <button role="button"> is redundant.`